src/app/content/hooks/locationChange/index.ts

```diff
@@ -14,7 +14,10 @@ const hookBody: RouteHookBody<typeof content> = (services) => {
   const boundRegisterPageView = registerPageView(services);
 
   return async(action) => {
-    await resolveContent(services, action.match);
+    const response = await resolveContent(services, action.match);
+    if (!response) {
+      return;
+    }
```

Comment:
@TomWoodward I'm unsure of the best way to do this but this is to prevent the intlHook from running.

Reply:
the only potentially cleaner thing would be to switch your resolveContent approach to throwing an error instead of dispatching and nooping the reset of the logic. then you could catch it here and return

Reply:
@TomWoodward I added a new error type above because BookNotFoundError would trigger a 404

Reply:
it won't if you catch it

Reply:
I'm catching it on line 21 - I don't understand how to get around this: https://github.com/openstax/rex-web/blob/main/src/app/utils.ts#L53